cache/util/fsutil.go

```diff
@@ -131,16 +131,23 @@ func StatFile(ctx context.Context, mount snapshot.Mountable, path string) (*fsty
 			// The filename here is internal to the mount, so we can restore
 			// the request base path for error reporting.
 			// See os.DirFS.Open for details.
-			err1 := err
-			if err := errors.Cause(err); err != nil {
-				err1 = err
-			}
-			if pe, ok := err1.(*os.PathError); ok {
-				pe.Path = path
-			}
+			setErrorPath(err, path)
 			return errors.WithStack(err)
 		}
 		return nil
 	})
 	return st, err
 }
+
+// setErrorPath will override the path in an os.PathError in the error chain.
+// This works with the fsutil library, but it isn't necessarily the correct
+// way to do this because the error message of wrapped errors doesn't necessarily
+// update or change when a wrapped error is changed.
+//
+// Still, this method of updating the path works with the way this specific
+// library returns errors.
+func setErrorPath(err error, path string) {
```

Comment:
Maybe `replaceErrorPath()` (or rewrite/overwrite) would make it more explicit that this is not adding a path to the chain but rewriting `PathError` properties if they are found.